### PR TITLE
tests: Add data-driven testcases to rpc_decodescript.py

### LINKED ISSUE #23641: Write more decodescript tests
There are some test cases mentioned in https://github.com/bitcoin/bitcoin/pull/23486#issue-1051148887 

Currently they need to be manually executed with bitcoin-cli. It would be nice to integrate them into the existing decodescript functional test.

I recommend to check the full response as opposed to just a partial response, as it is being done currently in the functional test.

#### Useful skills:

* Python3
* Background in our functional tests
* Basic background in our RPC interface
* Basic bitcoin script background

#### Want to work on this issue?

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by Kvaciral**
closes #23641

### CODE REVIEW

**Comment by jamesob** [comment]
Nice, concept ACK

**Comment by brunoerg** [comment]
Concept ACK

**Review by brunoerg — APPROVED** [review]
tACK f7ddfef9371f476533ddbbc1cf0c8520c01f8e42

**Review by alirezaayande — APPROVED** [review]

File: test/functional/rpc_decodescript.py [hunk 1]
```diff
@@ -252,13 +255,23 @@ def decoderawtransaction_asm_sighashtype(self):
         rpc_result = self.nodes[0].decoderawtransaction(txSave.serialize().hex())
         assert_equal('OP_RETURN 3011020701010101010101020601010101010101', rpc_result['vin'][0]['scriptSig']['asm'])
 
+    def decodescript_datadriven_tests(self):
+        with open(os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data/rpc_decodescript.json'), encoding='utf-8') as f:
+            dd_tests = json.load(f)
+
+        for dd_test in dd_tests:
```
Comment (MarcoFalke) [inline test/functional/rpc_decodescript.py:262]: nit: If you make `dd_tests` a dictionary of the form `{ 'input1': { ... rpc result ... }, 'input2': {...}, ...}`, you can avoid the array indices.

```suggestion
        for script, result in dd_tests:
```

**Review by maflcko — APPROVED** [review]
Thanks, ACK

File: test/functional/rpc_decodescript.py (hunk 1 above)
Comment (laanwj) [reply test/functional/rpc_decodescript.py:262]: I think that should work if you don't change the data type? `,` would unpack the arrays.
When converting to a dictionary you'd need to iterate over `dd_tests.items()`.

**Pull request merged**